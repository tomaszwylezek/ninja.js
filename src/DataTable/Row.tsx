import React, { FC } from "react";

export interface IRowProps {
  row: any;
}

const Row: FC<IRowProps> = props => {
  const { row } = props;

  return (
    <tr>
      <td>
        <a href={row.edit_path}>{row.name1}</a>
        <br />
        <small>{row.email}</small>
      </td>
    </tr>
  );
};

export default Row;
