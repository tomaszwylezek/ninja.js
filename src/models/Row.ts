export interface IRow {
  name1: string;
  email: string;
  edit_path: string;
  per_id: number;
}
