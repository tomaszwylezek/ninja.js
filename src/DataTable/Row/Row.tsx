import React, { FC } from 'react';

import { IRow } from '../../models';

export interface IRowProps {
  row: IRow;
}

export const Row: FC<IRowProps> = ({ row }) => (
  <tr>
    <td>
      <a href={row.edit_path}>{row.name1}</a>
      <br />
      <small>{row.email}</small>
    </td>
  </tr>
);
