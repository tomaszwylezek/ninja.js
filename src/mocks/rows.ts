import { IRow } from '../models';

export const getRows = (): IRow[] => [
  {
    name1: 'Mads L. Klausen',
    email: 'MadsLKlausen@jourrapide.com',
    edit_path: 'http://google.com',
    per_id: 1
  },
  {
    name1: 'Alfred K. Krogh',
    email: 'AlfredKKrogh@armyspy.com',
    edit_path: 'http://google.com',
    per_id: 2
  },
  {
    name1: 'Silas L. Bertelsen',
    email: 'SilasLBertelsen@armyspy.com',
    edit_path: 'http://google.com',
    per_id: 3
  },
  {
    name1: 'Mia A. Johnsen',
    email: 'MiaAJohnsen@dayrep.com',
    edit_path: 'http://google.com',
    per_id: 4
  },
  {
    name1: 'Alfred S. Schou',
    email: 'AlfredSSchou@jourrapide.com',
    edit_path: 'http://google.com',
    per_id: 5
  }
];
