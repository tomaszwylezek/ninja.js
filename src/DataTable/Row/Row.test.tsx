import React from 'react';
import { shallow } from 'enzyme';
import { Row, IRowProps } from './Row';
import { getRows } from '../../mocks';

describe('Row', () => {
  let props: IRowProps;

  beforeEach(() => {
    props = {
      row: getRows()[0]
    };
  });

  it('renders without crashing', () => {
    const component = shallow(<Row {...props} />);

    expect(component).toMatchSnapshot();
  });
});
