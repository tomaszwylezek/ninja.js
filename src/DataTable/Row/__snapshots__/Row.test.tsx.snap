// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders without crashing 1`] = `
<tr>
  <td>
    <a
      href="http://google.com"
    >
      Mads L. Klausen
    </a>
    <br />
    <small>
      MadsLKlausen@jourrapide.com
    </small>
  </td>
</tr>
`;
